packages/providers/src/resource.ts

```diff
@@ -25,6 +26,9 @@ export type RawMessage = {
 
 export type RawResource = RawCoin | RawMessage;
 export type Resource = Coin | MessageCoin;
+export type AccountResource<R extends Resource, T extends AbstractAccount> = R & {
```

Comment:
Ideally this would be defined wherever `Account` is defined, but then we'd have a problem with cyclic dependencies as `@fuels-ts/wallet`, where `Account` is defined, already depends on `@fuels-ts/providers`.
This type can easily be sorted out whenever the `interfaces` package's removal of the `AbstractAccount` happens, because that removal would entail a lot of reordering and this type would be caught in it.